app/features/settings/settings-sidebar.tsx

```diff
@@ -0,0 +1,72 @@
+import { Cog } from 'lucide-react';
+import { AnimatePresence, motion } from 'motion/react';
+import {
+  Sidebar,
+  SidebarProvider,
+  SidebarTrigger,
+} from '~/components/ui/sidebar';
+import {
+  SettingsSidebarProvider,
+  useSettingsSidebar,
+} from './settings-sidebar-provider';
+import { AddAccountView } from './views/add-account-view';
+import { AdvancedSettingsView } from './views/advanced-settings-view';
+import { AllAccountsView } from './views/all-accounts-view';
+import { AppearanceView } from './views/appearance-view';
+import { ContactsView } from './views/contacts-view';
+import { EditProfileView } from './views/edit-profile-view';
+import { MainView } from './views/main-view';
+import { QrView } from './views/qr-view';
+import { SingleAccountView } from './views/single-account-view';
+
+export function SettingsSidebarContent() {
+  const { state } = useSettingsSidebar();
+  const { view } = state;
+
+  return (
+    <Sidebar side="right">
+      <AnimatePresence mode="wait">
```

Comment:
is there no some kind of existing sidebar or drawer component that can be opened from the side with animation? i've seen such things in other component libs before and if there is it would be nice to avoid implementing this animtation ourselves

Reply:
or did I misunderstand this and sidebar is already animated while this framer motion things is here to animate views inside the sidebar?

Reply:
Yea the sidebar already has an animation when it opens. This creates the animations for the changing views